code-comment-style.md

```diff
@@ -0,0 +1,137 @@
+
+## Why a good comment matters?
+
+- To speed up the reviewing process
+- To help maintain the code
+- To improve the API document readability
+- To improve the development efficiency of the whole team
+
+## Where/When to comment?
+
+Write a comment where/when:
+
+- For important code
+- For obscure code
+- For tricky or interesting code
+- For a complex code block
+- If a bug exists in the code but you cannot fix it or you just want to ignore it for the moment
+- If the code is not optimal but you don’t have a smarter way now
+- To remind yourself or others of missing functionality or upcoming requirements not present in the code
+
+A comment is generally required for:
+
+- File
+- Type
+- Constant 
+- Function
+- Method
+- Variable
+- Typical algorithm
+- Exported name
+- Test case
+- TODO
+- FIXME
+
+## How to comment?
+
+### Format of a good comment (for Rust)
+
+- Non-doc comment
+        
+    - Use **//** for a line comment
+    - Use **/* … */** for a block comment (used only when needed)
+    
+- Doc comment
+        
+    - Use **///** for a line comment
+    - Use **//!** for a block comment
+    - Use **rustfmt** to format your code
+
+- Place the single-line and block comment above the code it’s annotating
+- Fold long lines of comments
+- Each line of text in your code and comment should be at most 100 characters long
+- Do not use relative URLs; use **absolute URLs** instead
+
+    Right:
+    
+    ```
+    // For more configuration details, see https://example.com/project/tools/config-items.  
+    ```
+    
+    Wrong:
+    
+    ```
+    // For more details, see [Configuration Items](../tools/config-items.md).
+    ```
+
+### Language for a good comment
+
+- Word
+    
+    - Use **American English** rather than British English
+        
+        - color, canceling, synchronize     (Recommended)
+        - colour, cancelling, synchronise   (Not recommended)
+    
+    - Use correct spelling
+
+    - Use **standard or official capitalization**
+        
+        - TiKV, TiDB-Binlog, Region, gRPC, RocksDB, GC, k8s, [mydumper](https://github.com/maxbube/mydumper), [Prometheus Pushgateway](https://github.com/prometheus/pushgateway)   (Right)
+        - Tikv, TiDB Binlog, region, grpc, rocksdb, gc, K8S, MyDumper, Prometheus PushGateway   (Wrong)
+
+    - Use words and expressions consistently
+        
+        - dead link vs. broken link (Only one of them can appear in a single document)
+    
+    - Do not use lengthy compound words
+
+    - Do not abbreviate unless it is absolutely necessary
+
+    - "we" can be used only when it means the code writer
+
+- Sentence
+
+    - Use standard grammar and correct punctuation
+    - Use relatively short sentences
+
+- For each comment, capitalize the first letter and end this sentence with a period
+    
+    - If a lower-case identifier comes at the beginning of a sentence, don't capitalize it
+
+    ```
+    // enterGame causes Players to enter the 

```

Comment:
For Rust, it is not recommended to use the function name as the start word in comments.

Reply:
Got it. I can add a note to it.  Here it does not specifically talk about function name. An identifier is a generic word that refers to any construction element like a variable, function, or method. In rust, can we use other names as the start words besides function?

Reply:
Will remove this since as confirmed by CC, this (identifier) is actually more intended for  Go.

Reply:
I would add that I would always use markdown code for names of functions, etc., so that if you *have* to start a sentence with a name, then it looks more reasonable, e.g.,

```
// `enter_game` causes Player to enter the ...
fn enter_game ...
```